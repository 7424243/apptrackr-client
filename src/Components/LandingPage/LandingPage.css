/* Classes */

.landing_section1_content {
    text-align: center;
}

.landing_section1 {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;
}
.landing {
    margin: auto;
    width: 70%;
}

.landing_table {
    margin: 15px;
}


@media
only screen and (max-width: 625px) {
  table, thead, tbody, th, td, tr {
    display: block;
  }
  thead tr {
    position: absolute;
    top: -9999px;
    left: -9999px;
  }
  tr { border: 1px solid #ccc; }
  td {
    border: none;
    border-bottom: 1px solid #eee;
    position: relative;
    padding-left: 200px;
    margin-left: 125px;
    overflow-wrap: normal;
  }
  td:before {
    position: absolute;
    top: 12px;
    left: 6px;
    width: 200px;
    padding-right: 40px;
    white-space: nowrap;
    margin-left: -125px;
  }
  td:nth-of-type(1):before { content: "Job:"; }
  td:nth-of-type(2):before { content: "Company:"; }
  td:nth-of-type(3):before { content: "Date Applied:"; }
  td:nth-of-type(4):before { content: "Interview Date:";}
  td:nth-of-type(5):before { content: "Status:";}
}