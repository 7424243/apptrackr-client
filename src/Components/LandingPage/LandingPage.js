import React from 'react'
import {NavLink} from 'react-router-dom'
import './LandingPage.css'

function LandingPage() {
    return (
        <>
            <section className='landing_section1'>
            <div className='landing_section1_content'>
                <div className='landing'>
                    <h2>Easily organize job applications</h2>
                    <p>Keep track of details for each job application</p>
                    <p>Save frequently utilized resources for convenient access</p>
                    <p><NavLink to={'/login'}>Login</NavLink> with a demo account to take a peak</p>
                    <p>Username: demo</p>
                    <p>Password: Demo1234!</p>
                </div>
            </div>
            </section>
            <div className='landing_table'>
                <table>
                    <thead>
                        <tr>
                            <th>Job</th>
                            <th>Company</th>
                            <th>Date Applied</th>
                            <th>Interview Date</th>
                            <th className='last_th'>Status</th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr>
                            <td>{'Full Stack Software Engineer'}</td>
                            <td>{'Olive'}</td>
                            <td></td>
                            <td></td>
                            <td className='last_td'>{'Interested'}</td>
                        </tr>
                        <tr>
                            <td>{'EDU Software Engineer'}</td>
                            <td>{'Splunk'}</td>
                            <td>{'02/03/2021'}</td>
                            <td>{'02/10/2021'}</td>
                            <td className='last_td'>{'Applied'}</td>
                        </tr>
                        <tr>
                            <td>{'Software Engineer'}</td>
                            <td>{'Google'}</td>
                            <td>{'02/15/2021'}</td>
                            <td>{'03/01/2021'}</td>
                            <td className='last_td'>{'Applied'}</td>
                        </tr>
                        <tr>
                            <td>{'Web Developer'}</td>
                            <td>{'Toddy LLC'}</td>
                            <td>{'02/15/2021'}</td>
                            <td>{'02/26/2021'}</td>
                            <td className='last_td'>{'Closed'}</td>
                        </tr>
                    </tbody>
                </table>
            </div>
            
        </>
    )
}

export default LandingPage