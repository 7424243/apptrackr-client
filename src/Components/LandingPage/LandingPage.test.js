import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom'
import LandingPage from './LandingPage';

describe('LandingPage Component', () => {

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><LandingPage/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter><LandingPage/></BrowserRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})