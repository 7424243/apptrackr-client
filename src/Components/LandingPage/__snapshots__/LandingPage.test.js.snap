// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component renders the UI as expected 1`] = `
Array [
  <section
    className="landing_section1"
  >
    <div
      className="landing_section1_content"
    >
      <div
        className="landing"
      >
        <h2>
          Easily organize job applications
        </h2>
        <p>
          Keep track of details for each job application
        </p>
        <p>
          Save frequently utilized resources for convenient access
        </p>
        <p>
          View a snapshot of the main page below
        </p>
        <p>
          <a
            aria-current={null}
            className="landing_login"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
           with demo credentials to sample full functionality of the app
        </p>
      </div>
    </div>
  </section>,
  <div
    className="landing_table"
  >
    <table>
      <thead>
        <tr>
          <th>
            Job
          </th>
          <th>
            Company
          </th>
          <th>
            Date Applied
          </th>
          <th>
            Interview Date
          </th>
          <th
            className="last_th"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Full Stack Software Engineer
          </td>
          <td>
            Olive
          </td>
          <td>
            --
          </td>
          <td>
            --
          </td>
          <td
            className="last_td"
          >
            Interested
          </td>
        </tr>
        <tr>
          <td>
            EDU Software Engineer
          </td>
          <td>
            Splunk
          </td>
          <td>
            02/03/2021
          </td>
          <td>
            02/10/2021
          </td>
          <td
            className="last_td"
          >
            Applied
          </td>
        </tr>
        <tr>
          <td>
            Software Engineer
          </td>
          <td>
            Google
          </td>
          <td>
            02/15/2021
          </td>
          <td>
            --
          </td>
          <td
            className="last_td"
          >
            Applied
          </td>
        </tr>
        <tr>
          <td>
            Web Developer
          </td>
          <td>
            Toddy LLC
          </td>
          <td>
            02/15/2021
          </td>
          <td>
            02/26/2021
          </td>
          <td
            className="last_td"
          >
            Closed
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
